kotlin-checks-test-sources/src/main/kotlin/checks/SimplifySizeExpressionCheckSample.kt

```diff
@@ -0,0 +1,236 @@
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        if (list != null && 0 < list.size) foo() // Noncompliant {{Replace null check and collection size check with "!isNullOrEmpty()"}}
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        if (list != null && list.size == 0) foo() // Noncompliant {{Replace collection size check with "isEmpty()"}}
+//                          ^^^^^^^^^^^^^^
+
+        if (list == null && list2.size != 0) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+//                          ^^^^^^^^^^^^^^^
+        if (list == null && list2.size > 0) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+//                          ^^^^^^^^^^^^^^
+        if (list == null && 0 < list2.size) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+//                          ^^^^^^^^^^^^^^
+        if (list == null && list2.size == 0) foo() // Noncompliant {{Replace collection size check with "isEmpty()"}}
+//                          ^^^^^^^^^^^^^^^
+
+        if (list2.size != 0 && list2 != null) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+//          ^^^^^^^^^^^^^^^
+        if (list2.size == 0 && list2 != null) foo() // Noncompliant {{Replace collection size check with "isEmpty()"}}
+        if (list2.size != 0 && list2 == null) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+        if (list2.size == 0 && list2 == null) foo() // Noncompliant {{Replace collection size check with "isEmpty()"}}
+
+        if (list == null || list.size != 0) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+//                          ^^^^^^^^^^^^^^
+        if (list == null || list.size > 0) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+//                          ^^^^^^^^^^^^^
+        if (list == null || 0 == list.size) foo() // Noncompliant {{Replace null check and collection size check with "isNullOrEmpty()"}}
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        if (list == null || list.size == 0) foo() // Noncompliant {{Replace null check and collection size check with "isNullOrEmpty()"}}
+        if (0 == list2.size || list == null) foo() // Noncompliant {{Replace collection size check with "isEmpty()"}}
+//          ^^^^^^^^^^^^^^^
+        if (list2.size != 0 || list == null) foo() // Noncompliant {{Replace collection size check with "isNotEmpty()"}}
+
+        if (list != null && list.isNotEmpty()) foo() // Noncompliant {{Replace null check and collection size check with "!isNullOrEmpty()"}}
+        if (list != null && list.isEmpty()) foo() // Compliant
+        if (list == null && list2.isNotEmpty()) foo() // Compliant
+        if (list == null && list2.isEmpty()) foo() // Compliant
+
+        if (list == null || list.isEmpty()) foo() // Noncompliant {{Replace null check and collection size check with "isNullOrEmpty()"}}
```

Comment:
have you checked `and` and `or` in this examples

Reply:
Do you mean the bitwise operators? But they operate on `Int` and `Long` only in Kotlin, not on `Boolean`.